features/steps/givens/attributes.py

```diff
@@ -68,12 +68,16 @@ def step_impl(context, inst, attribute, tail="single"):
 @gherkin_ifc.step("Its {attribute} attribute {condition} with {prefix}")
 def step_impl(context, inst, attribute, condition, prefix):
     assert condition in ('starts', 'does not start')
+    attr = str(getattr(inst, attribute, ''))
     if condition == 'starts':
-        if hasattr(inst, attribute) and str(getattr(inst, attribute, '')).startswith(prefix):
+        if attr and attr.startswith(prefix):
+            context.activation_attr = attr
```

Comment:
I don't think this works. Isn't this one global variable for  the entire model? Did you test with multiple psets in one file?

Reply:
I think you're right. It did work with 
![Screenshot from 2024-05-12 17-23-09](https://github.com/buildingSMART/ifc-gherkin-rules/assets/54070862/3dffe82f-8c1e-4373-9ec1-c5dde253ef5a)
But that's only because they have the same pset name. But when the names are different it's overwritten before the next statement is reached.